lib/runtime/src/utils/worker_monitor.rs

```diff
@@ -0,0 +1,190 @@
+// SPDX-FileCopyrightText: Copyright (c) 2024-2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// TODO: Make load comparisons and runtime metrics a generic trait so this monitoring
+// system is not tied to KV cache concepts, which are LLM-specific. This would allow
+// different types of workers to define their own load metrics and busy thresholds.
+
+use crate::component::{Client, InstanceSource};
+use crate::traits::events::EventSubscriber;
+use crate::traits::DistributedRuntimeProvider;
+use crate::utils::typed_prefix_watcher::{key_extractors, watch_prefix_with_extraction};
+use std::collections::HashMap;
+use std::sync::{Arc, RwLock};
+use tokio::sync::watch;
+use tokio_stream::StreamExt;
+
+// Constants for monitoring configuration
+const KV_METRICS_SUBJECT: &str = "kv_metrics";
+const MODEL_ROOT_PATH: &str = "models";
+
+// Internal structs for deserializing metrics events
+#[derive(serde::Deserialize)]
+struct LoadEvent {
+    worker_id: i64,
+    data: ForwardPassMetrics,
+}
+
+#[derive(serde::Deserialize)]
+struct ForwardPassMetrics {
```

Comment:
Why do we have a duplicate definition of ForwardPassMetrics (and possibly other structs) here? @PeaBrane